src/engine/objects/robot/mechanisms/SimGripperMechanism.ts

```diff
@@ -0,0 +1,335 @@
+      );
+    }
+
+    // Get mount positions
+    const sensorPosition = getSensorMountPosition(
+      robotSpec,
+      spec.mountFace,
+      spec.mountOffset
+    );
+
+    // Backboard object
+    let backboardDimensionX = 0;
+    let backboardDimensionZ = 0;
+
+    let range = 0.1;
+    let backboard = 0.01;
+    let jawThickness = 0.01;
+    let width = 0.4;
+
+    this._spec = spec;
+    this._range = range;
+    this._width = width;
+    this._closeSpeed = 0.1;
+    this._openSpeed = -0.1;
+    this._robot = robot;
+    this._state = GripperState.OPENING;
+
+    switch (spec.mountFace) {
+      case SensorMountingFace.LEFT:
+      case SensorMountingFace.RIGHT:
+        backboardDimensionX = backboard;
+        backboardDimensionZ = width;
+        break;
+      case SensorMountingFace.FRONT:
+      case SensorMountingFace.REAR:
+        backboardDimensionX = width;
+        backboardDimensionZ = backboard;
+        break;
+    }
+
+    this._bodySpecs = {
+      type: "dynamic", // sensors are always dynamic
+      position: new Vec2(sensorPosition.x, sensorPosition.z),
+      angle: 0,
+    };
+    this._fixtureSpecs = {
+      shape: new Box(backboardDimensionX / 2, backboardDimensionZ / 2),
+      isSensor: false,
+      density: 1,
+      filterCategoryBits: EntityCategory.ROBOT_PART,
+      filterMaskBits: EntityMask.ROBOT_PART,
+    };
+    let backboardGeom = new THREE.BoxGeometry(
+      backboardDimensionX,
+      robotSpec.dimensions.y,
+      backboardDimensionZ
+    );
+    // Create meshes
+    this._mesh = new THREE.Mesh(
+      backboardGeom,
+      new THREE.MeshStandardMaterial({ color: 0x003377 })
+    );
+
+    // Create jaws
+    this._jaws = [
+      new SimGripperJaw(jawThickness, range, -width / 2, spec, robotSpec),
+      new SimGripperJaw(jawThickness, range, width / 2, spec, robotSpec),
+    ];
+
+    this._jaws.forEach(this.addChild.bind(this));
+  }
+
+  public getValue(ioIdentifier: string): number {
+    if (ioIdentifier !== IO_TYPE_HELD) {
+      throw new Error("Unsupported operation");
+    }
+
+    return this._state == GripperState.CLOSED_FULL ? 1.0 : 0.0;
+  }
+
+  public setValue(ioIdentifier: string, value: number | boolean): void {
+    if (ioIdentifier !== IO_TYPE_GRAB) {
+      throw new Error("Unsupported operation");
+    }
+
+    enum Action {
+      CLOSE,
+      OPEN,
+    }
+
+    let action = value >= 0.5 ? Action.CLOSE : Action.OPEN;
+
+    console.log("Gripper setValue", value, action);
+    switch (action) {
+      case Action.OPEN:
+        this._state = GripperState.OPENING;
+        this._slideJoint.setMotorSpeed(this._closeSpeed);
+        break;
+      case Action.CLOSE:
+        this._state = GripperState.CLOSING;
+        this._slideJoint.setMotorSpeed(this._openSpeed);
+        break;
+    }
+  }
+
+  public configureFixtureLinks(world: World) {
+    // link jaw bodies to backboard
+    let fixedJaw = this._jaws[0];
+    let slideJaw = this._jaws[1];
+
+    world.createJoint(
```

Comment:
You could create a second prismatic joint here for the other jaw too. Which would enable you to have the jaws centered when fully closed

Reply:
I think its a good idea, I did things this way just to get things working. though I dont think there is anything preventing us from doing it the way you suggest. though ill do that in a different PR.